File: JAVASCRIPT/ComposicionAgregacion/Clase 30_09/JS/Cuenta.js
class Cuenta {

    #nroCuenta;
    #tipoCuenta;
    #saldo;

    constructor(nroCuenta, tipoCuenta, saldo) {
        this.#nroCuenta = nroCuenta;
        this.#tipoCuenta = tipoCuenta;
        this.#saldo = saldo;
    }

    getNroCuenta(){
        return this.#nroCuenta;
    }

    setNroCuenta(nroCuenta){
        this.#nroCuenta = nroCuenta;
    }

    getTipoCuenta(){
        return this.#tipoCuenta;
    }
    
    setTipoCuenta(tipoCuenta){
        this.#tipoCuenta = tipoCuenta;
    }

    getSaldo(){
        return this.#saldo;
    }

    setSaldo(saldo){
        this.#saldo = saldo;
    }

    toString(){
        return `Cuenta [numero de cuenta: ${this.#nroCuenta}, tipo Cuenta: ${this.#tipoCuenta}. saldo: ${this.#saldo}]`;
    }

    consignar(nroCuenta, monto){
        let flag = false;
        
        if (nroCuenta === "1234") {
            this.#saldo += monto;
            flag = true;
        }
        return flag;
    }

    consignarMsg(nroCuenta,monto){
        let msg = "";
        if (nroCuenta === "1234") {
            this.#saldo += monto;
            msg = "Transacción exitosa...." + "saldo actual: " + this.#saldo;
        }else{
            msg = "Error en la transacción..."
        }

        return msg;
    }
}

export default Cuenta;

File: 20242/Clase 30_09/ComposicionAgregracion/Clase/Cuenta.java
public class Cuenta {
    

    private String nroCuenta;
    private String tipoCuenta;
    private double Saldo;

    
    public Cuenta(String nroCuenta, String tipoCuenta, double Saldo) {
        this.nroCuenta = nroCuenta;
        this.tipoCuenta = tipoCuenta;
        this.Saldo = Saldo;
    }


    public String getNroCuenta() {
        return nroCuenta;
    }


    public void setNroCuenta(String nroCuenta) {
        this.nroCuenta = nroCuenta;
    }


    public String getTipoCuenta() {
        return tipoCuenta;
    }


    public void setTipoCuenta(String tipoCuenta) {
        this.tipoCuenta = tipoCuenta;
    }


    public double getSaldo() {
        return Saldo;
    }


    public void setSaldo(double saldo) {
        Saldo = saldo;
    }


    @Override
    public String toString() {
        return "Cuenta [nroCuenta=" + nroCuenta + ", tipoCuenta=" + tipoCuenta + "]";
    }


    
}
